src/query/catalog/src/catalog/session_catalog.rs

```diff
@@ -357,43 +356,28 @@ impl Catalog for SessionCatalog {
         self.inner.upsert_table_option(tenant, db_name, req).await
     }
 
-    async fn update_table_meta(
+    async fn retryable_update_multi_table_meta(
         &self,
-        table_info: &TableInfo,
-        req: UpdateTableMetaReq,
-    ) -> Result<UpdateTableMetaReply> {
-        let state = self.txn_mgr.lock().state();
-        match state {
-            TxnState::AutoCommit => self.inner.update_table_meta(table_info, req).await,
-            TxnState::Active => {
-                self.txn_mgr.lock().update_table_meta(req, table_info);
-                Ok(UpdateTableMetaReply {
-                    share_table_info: None,
-                })
-            }
-            TxnState::Fail => unreachable!(),
+        req: UpdateMultiTableMetaReq,
+    ) -> Result<UpdateMultiTableMetaResult> {
+        if req
+            .update_table_metas
+            .iter()
+            .any(|(_, info)| matches!(info.db_type, DatabaseType::ShareDB(_)))
```

Comment:
SessionCatalog should not care aboue database type, it will be impl be inner catalog.